lisa/analysis/base.py

```diff
@@ -152,16 +152,31 @@ def save_plot(self, figure, filepath=None, img_format=None):
         default_dir = '.'
         return self._save_plot(figure, default_dir, filepath, img_format)
 
-    def _do_plot(self, plotter, filepath, axis):
+    def do_plot(self, plotter, filepath, axis, **kwargs):
         """
         Simple helper for consistent behavior across methods.
+
+        :returns: An :class:`matplotlib.axes.Axes` containing the plot.
+
+        :param filepath: Path of the file to save the figure in. If `None`, no
+            file is saved.
+        :type filepath: str or None
+
+        :param axis: instance of :class:`matplotlib.axes.Axes` to plot into.
+            If `None`, a new figure and axis are created and returned.
+        :type axis: matplotlib.axes.Axes
+            or numpy.ndarray(matplotlib.axes.Axes)
+            or None
+
+        :param kwargs: keyword arguments forwarded to :meth:`setup_plot`
+        :type kwargs: dict
         """
 
         local_fig = not axis
```

Comment:
That doesn't work anymore for multi-axis plots:
```
The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

We probably want `axis is None`